qiskit/transpiler/preset_passmanagers/level3.py

```diff
@@ -129,6 +134,7 @@ def _opt_control(property_set):
     if coupling_map:
         pm3.append(_given_layout)
         pm3.append(_choose_layout, condition=_choose_layout_condition)
+        pm3.append(_remove_gates, condition=_remove_gates_condition)
```

Comment:
I think the best place for this is after `_unroll` and before the `if coupling_map` check, since we can apply these optimizations even without a `coupling_map`, and they won't impact the layout selection process (they might improve it slightly).